### ISSUE TITLE
LibWeb: Regression in block-level element height calculation after margin collapsing

### ISSUE THREAD

**Issue by itamar8910** [SerenityOS/serenity#17120]
There has been a regression in the rendering of chessboards in https://chessboardjs.com.

I bisected it to fc180563ea79873e0113334c2db8c54f98184044 (#16632 - LibWeb: Implement margin collapsing between parent and first child).

Minimal example that reproduces the issue:
``` html
<!DOCTYPE html>
<html><head>
    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
          <style>
            .clearfix-7da63{clear:both}
            .square-55d63{float:left;}
            .white-1e1d7{background-color:#f0d9b5;color:#b58863}
            .black-3c85d{background-color:#b58863;color:#f0d9b5}
          </style>
      </head>
       <body>
         <div style="width: 392px;">
            <div id="div1">
               <div class="square-55d63 white-1e1d7 " style="width:49px;height:49px;"></div>
               <div class="clearfix-7da63"></div>
            </div>
            <div>
               <div class="square-55d63 black-3c85d " style="width:49px;height:49px;" ></div>
            </div>
         </div>
    </body>
</html>
```
Left: before regression
Right: after regression
![image](https://user-images.githubusercontent.com/9247514/213871174-3ce05b40-6181-4339-a80a-6e8df0038e2a.png) ![image](https://user-images.githubusercontent.com/9247514/213871139-a2851b2d-24ad-47a4-83da-2e893906b8d5.png)  

After the regression, the height of `div1` is calculated to be 0.

**Comment by ADKaster** [comment]
CC @kalenikaliaksandr

**Comment by awesomekling** [comment]
This was fixed in #17125 :)
